ob1k-cache/src/main/java/com/outbrain/ob1k/cache/LoadingCacheDelegate.java

```diff
@@ -98,37 +106,37 @@ public void produce(final Consumer<V> consumer) {
                     return;
                 }
 
-                final ComposableFuture<V> cachedResult = cache.getAsync(key).withTimeout(duration, timeUnit, "LoadingCacheDelegate fetch from cache");
+                final ComposableFuture<V> cachedResult = cache.getAsync(key)
+                  .continueOnError(handleCacheErrors(promise, key))
```

Comment:
I see three problems here:

1. you set the timeout handler after the error handler which means that it will miss the timeoutException
2. you have duplicate handling of errors from the cache, both in handleCacheErrors() and inside the consumer of the cache future result
3. you're adding code to blocks that update metrics which means that they won't get called when there are no metrics defined.